Detectors/ITSMFT/ITS/tracking/src/VertexerTraits.cxx

```diff
@@ -228,42 +232,46 @@ const std::vector<std::pair<int, int>> VertexerTraits::selectClusters(const std:
 
 void VertexerTraits::computeTrackletsPureMontecarlo()
 {
-
-  std::cout << "Running in Montecarlo trivial mode\n";
+  assert(mEvent != nullptr);
 
   std::vector<int> foundTracklets01;
   std::vector<int> foundTracklets12;
 
-  std::vector<int> labelsMC0 = getMClabelsLayer(0);
-  std::vector<int> labelsMC1 = getMClabelsLayer(1);
-  std::vector<int> labelsMC2 = getMClabelsLayer(2);
-
   for (unsigned int iCurrentLayerClusterIndex{ 0 }; iCurrentLayerClusterIndex < mClusters[0].size(); ++iCurrentLayerClusterIndex) {
-    auto& cluster{ mClusters[0][iCurrentLayerClusterIndex] };
+    auto& currentCluster{ mClusters[0][iCurrentLayerClusterIndex] };
     for (unsigned int iNextLayerClusterIndex = 0; iNextLayerClusterIndex < mClusters[1].size(); iNextLayerClusterIndex++) {
       const Cluster& nextCluster{ mClusters[1][iNextLayerClusterIndex] };
-      if (labelsMC1[iNextLayerClusterIndex] == labelsMC0[iCurrentLayerClusterIndex] && labelsMC1[iNextLayerClusterIndex] != -1)
-        mComb01.emplace_back(iCurrentLayerClusterIndex, iNextLayerClusterIndex, cluster, nextCluster);
+      const auto& lblNext = mEvent->getClusterLabels(1, nextCluster.clusterId);
+      const auto& lblCurr = mEvent->getClusterLabels(0, currentCluster.clusterId);
+      if (lblNext.getTrackID() == lblCurr.getTrackID() && lblCurr.isValid()) {
+        mComb01.emplace_back(iCurrentLayerClusterIndex, iNextLayerClusterIndex, currentCluster, nextCluster);
+      }
     }
   }
 
   for (unsigned int iCurrentLayerClusterIndex{ 0 }; iCurrentLayerClusterIndex < mClusters[2].size(); ++iCurrentLayerClusterIndex) {
-    auto& cluster{ mClusters[2][iCurrentLayerClusterIndex] };
+    auto& currentCluster{ mClusters[2][iCurrentLayerClusterIndex] };
     for (unsigned int iNextLayerClusterIndex = 0; iNextLayerClusterIndex < mClusters[1].size(); iNextLayerClusterIndex++) {
       const Cluster& nextCluster{ mClusters[1][iNextLayerClusterIndex] };
-      if (labelsMC1[iNextLayerClusterIndex] == labelsMC2[iCurrentLayerClusterIndex] && labelsMC1[iNextLayerClusterIndex] != -1)
-        mComb12.emplace_back(iNextLayerClusterIndex, iCurrentLayerClusterIndex, nextCluster, cluster);
+      const auto& lblNext = mEvent->getClusterLabels(1, nextCluster.clusterId);
+      const auto& lblCurr = mEvent->getClusterLabels(2, currentCluster.clusterId);
+      if (lblNext.getTrackID() == lblCurr.getTrackID() && lblCurr.isValid()) {
+        mComb12.emplace_back(iNextLayerClusterIndex, iCurrentLayerClusterIndex, nextCluster, currentCluster);
+      }
     }
   }
 
   for (auto& trklet01 : mComb01) {
     for (auto& trklet12 : mComb12) {
       if (trklet01.secondClusterIndex == trklet12.firstClusterIndex) {
         const float deltaTanLambda{ gpu::GPUCommonMath::Abs(trklet01.tanLambda - trklet12.tanLambda) };
```

Comment:
@mconcas : this gives me a build error with -Werror (apparently the CI didn't catch it, probably the cmake migration broke this, investigating). Anyway, can you move the definition inside the `#if defined` protection?